packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -41,11 +43,13 @@ export interface DayCalendarSlots<TDate extends PickerValidDate> {
 }
 
 export interface DayCalendarSlotProps<TDate extends PickerValidDate> {
-  day?: SlotComponentPropsFromProps<
-    PickersDayProps<TDate>,
-    {},
-    DayCalendarProps<TDate> & { day: TDate; selected: boolean }
-  >;
+  day?: SlotComponentPropsFromProps<PickersDayProps<TDate>, {}, PickersDayOwnerState<TDate>>;
+}
+
+export interface PickersDayOwnerState<TDate extends PickerValidDate> extends PickerOwnerState {
```

Comment:
That one is up for discussion.

I also think this object should eventually be the one that is passed to the `styleOverrides` that @noraleonte is re-working (so that all customization API around day have the same `ownerState`).
So I'm only adding things that for me will obviously be in the one @noraleonte is working on.

Reply:
The approach makes sense to me. 👍 
Could we just align on naming? 🙈 
```suggestion
export interface PickerDayOwnerState<TDate extends PickerValidDate> extends PickerOwnerState {
```